Apromore-Core-Components/Apromore-Portal/src/main/java/org/apromore/portal/security/helper/JwtHelper.java

```diff
@@ -72,23 +85,50 @@ public static String readCookie(final HttpServletRequest httpServletRequest,
         return null;
     }
 
+    public static void writeCookie(final HttpServletResponse httpServletResponse,
+                                   final String cookieName,
+                                   final String cookieValue) {
+        final Cookie cookie = new Cookie(cookieName, cookieValue);
+        cookie.setPath("/");
+        httpServletResponse.addCookie(cookie);
+        LOGGER.debug("Added written cookie {} with name {} and value {} to the HttpServletResponse",
+                cookie, cookie.getName(), cookie.getValue());
+    }
+
+    public static JWTClaimsSet refreshJwt(final JWTClaimsSet jwtClaimsSet) {
+        try { Thread.sleep(10); } catch (final InterruptedException ie) { };
+
+        final Instant newExpiryTime = Instant.now().plus(WEBAPP_SSO_SESSION_TIMEOUT);
+
+        final JWTClaimsSet updatedJwtClaimsSet = new JWTClaimsSet.Builder()
+                .claim(JWT_KEY_SUBJECT_USERNAME, jwtClaimsSet.getSubject())
+                .claim(JWT_KEY_SUBJECT_EMAIL, jwtClaimsSet.getClaim(JWT_KEY_SUBJECT_EMAIL))
+                .claim(JWT_KEY_GIVEN_NAME, jwtClaimsSet.getClaim(JWT_KEY_GIVEN_NAME))
+                .claim(JWT_KEY_FAMILY_NAME, jwtClaimsSet.getClaim(JWT_KEY_FAMILY_NAME))
+                .claim(JWT_KC_USER_ID, jwtClaimsSet.getClaim(JWT_KC_USER_ID))
+                .claim(JWT_KEY_ISSUED_AT, jwtClaimsSet.getIssueTime())
+                .claim(JWT_EXPIRY_TIME, (newExpiryTime.toEpochMilli() / 1000)) // JWT spec is epoch *seconds* relative
+                .claim("str" + JWT_KEY_ISSUED_AT, jwtClaimsSet.getClaim("str" + JWT_KEY_ISSUED_AT))
+                .claim("str" + JWT_EXPIRY_TIME, jwtClaimsSet.getClaim("str" + JWT_EXPIRY_TIME))
+                .build();
+
+        return updatedJwtClaimsSet;
+    }
+
     public static boolean isJwtExpired(final JWTClaimsSet jwtClaimsSet,
-                                       final String issuedAtStr,
-                                       final String expiredAtStr) {
-        final long jwtExpiryEpoch;
-        if (expiredAtStr != null) {
-            jwtExpiryEpoch = Long.valueOf(expiredAtStr.substring(3));
-        } else { // Calculate relative
-            final Long jwtIssuedAt = Long.valueOf(issuedAtStr.substring(3));
-
-            jwtExpiryEpoch = calculateExpiryTime(jwtIssuedAt, WEBAPP_SSO_SESSION_TIMEOUT);
-        }
+                                       final Date issuedAtDate,
+                                       final Date expiryAtDate) {
```

Comment:
issuedAtDate and expiryAtDate are unused.

Reply:
Removed these two unused params.